src/main/java/org/sonar/plugins/stash/StashRequestFacade.java

```diff
@@ -141,39 +143,23 @@ public void postSonarQubeReport(PullRequestRef pr, String sonarQubeURL, SonarQub
   }
 
   /**
-   * Push Code Coverage report into the pull-request as comments.
-   */
-  public void postCoverageReport(PullRequestRef pr, String sonarQubeURL, CoverageIssuesReport coverageReport, StashDiffReport diffReport, StashClient stashClient) {
-    try {
-      if (!coverageReport.isEmpty()) {
-
-        postCommentPerIssue(pr, sonarQubeURL, coverageReport.getLoweredIssues(), diffReport, stashClient);
-
-        LOGGER.info("Code coverage report has been reported to Stash.");
-      }
-    } catch (StashClientException e) {
-      LOGGER.error("Unable to push code coverage report to Stash: {}", e.getMessage());
-      LOGGER.debug(STACK_TRACE, e);
-    }
-  }
-
-  /**
    * Post one comment by found issue on Stash.
    */
-  void postCommentPerIssue(PullRequestRef pr, String sonarQubeURL,
+  void postCommentPerIssue(PullRequestRef pr,
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3776)